Source/WebCore/platform/audio/ios/MediaSessionHelperIOS.mm

```diff
@@ -249,10 +249,10 @@ - (void)stopMonitoringAirPlayRoutes;
     END_BLOCK_OBJC_EXCEPTIONS
 }
 
-void MediaSessionHelperiOS::providePresentingApplicationPID(int pid)
+void MediaSessionHelperiOS::providePresentingApplicationPID(int pid, ShouldOverride shouldOverride)
 {
 #if HAVE(CELESTIAL)
-    if (m_presentedApplicationPID)
+    if (m_presentedApplicationPID && (*m_presentedApplicationPID == pid || shouldOverride == ShouldOverride::No))
```

Comment:
This seems unnecessary. Why do we need the `shouldOverride` flag? Can't we just do something like:

```
if (m_presentedApplicationPID == pid)
    return;
```

Reply:
It seems like there's a bunch of missing logic here. Something in the GPU process needs to decide which of the GPUConnectionToWebProcess should provide the "presented application PID". Right now, each connection just "steals" the presentingApplicationPID setting whenever it starts capture. That _kind of_ works now but only because of policy. It seems like there should be some code inside GPUProcess that is tracking which WebContent process is capturing and doing some kind of MRU tracking.

Reply:
> This seems unnecessary. Why do we need the `shouldOverride` flag?

It reduces the potential impact of this patch to capture only.
I agree we should remove shouldOverride at some point but this does not seem to be the right time.

> Right now, each connection just "steals" the presentingApplicationPID setting whenever it starts capture. That _kind of_ works now but only because of policy.

Capture can only happen one page/one process at a time, this is enforced at UIProcess level.
If we change this policy, this patch would break but many things would break at the same time.

This patch is a temporary workaround.
The long term plan is to have the capture session instance able to track which application it is related to with an audit token (or something similar). This patch would then become moot when we will have this.

Reply:
Okay, lets prioritize cleaning this up soon after this workaround lands.